### PR TITLE
updated joining docs

### PR DESCRIPTION
**Pull request by nickeubank**

### CODE REVIEW

File: doc/source/mergingdata.rst [hunk 1]
```diff
@@ -1,15 +1,73 @@
...
+   countries.head()
+   cities.head()
+
+   # Execute spatial join
+   from geopandas.tools import sjoin
+   cities_with_country = sjoin(cities, countries, how="inner", op='intersects')
+   cities_with_country.head()
+
+
+The ``op`` options determines the type of join operation to apply. ``op`` can be set to "intersects", "within" or "contains" (these are all equivalent when joining points to polygons, but different substantially when joining polygons to other polygons or lines).
```
Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: "different substantially" -> "differ substantially" or "substantially different" ?

Comment (nickeubank) [inline doc/source/mergingdata.rst]: tweaked, thanks

**Comment by nickeubank** [comment]
@jorisvandenbossche rebased for new data

File: doc/source/mergingdata.rst [hunk 2]
```diff
@@ -1,15 +1,73 @@
...
+
+In an attribute join, a ``GeoSeries`` or ``GeoDataFrame`` is combined with a regular *pandas* ``Series`` or ``DataFrame`` based on a common variable. This is analogous to normal merging or joining in *pandas*.
+
+In a Spatial Join, observations from to ``GeoSeries`` or ``GeoDataFrames`` are combined based on their spatial relationship to one another. 
 
 Attribute Joins
 ----------------
 
-[TO BE COMPLETED -- EXAMPLES OF JOINING GDF WITH PANDAS DATAFRAME]
+Attribute joins are accomplished using the ``merge`` method. Note that unlike in *pandas*, ``merge`` is a method called from the spatial dataset, not a stand-alone function.
```
Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: > Note that unlike in *pandas*, ``merge`` is a method called from the spatial dataset, not a stand-alone function

This is not fully correct, as in pandas you have both the method and the top-level function. So I would just emphasize that you need to use the method (to have a GeoDataFrame as the result)

Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: Actually, just tested it, and with `pd.merge` it results in a GeoDataFrame as well, if the GeoDataFrame is the left argument

File: doc/source/mergingdata.rst [hunk 3]
```diff
@@ -1,15 +1,73 @@
...
+   country_shapes = country_shapes.merge(country_names, on='iso_a3')
+   country_shapes.head()
+
 
 
 Spatial Joins
 ----------------
 
-[TO BE COMPLETED -- EXAMPLES OF SPATIAL JOINS]
+In a Spatial Join, to ``Geo`` objects are merged based on their spatial relationship to one another.
```
Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: "``Geo``" -> I would just speak about "geometry objects"

File: doc/source/mergingdata.rst [hunk 4]
```diff
@@ -1,15 +1,73 @@
...
+
+   # Execute spatial join
+   from geopandas.tools import sjoin
+   cities_with_country = sjoin(cities, countries, how="inner", op='intersects')
+   cities_with_country.head()
+
+
+The ``op`` options determines the type of join operation to apply. ``op`` can be set to "intersects", "within" or "contains" (these are all equivalent when joining points to polygons, but differ when joining polygons to other polygons or lines). 
+
+Note more complicated spatial relationships can be studying by combining geometric operations with spatial join. To find all polygons within a given distance of a point, for example, one can first use the ``buffer`` method to expand each point into a circle of appropriate radius, then intersect those buffered circles with the polygons in question.
```
Comment (jorisvandenbossche) [inline doc/source/mergingdata.rst]: "can be studying" -> "can be studied"

**Comment by jorisvandenbossche** [comment]
@nickeubank Few comments, but for the rest looks good. Thanks a lot!

Further, a general comment: I personally prefer limiting the comments inside `ipython` code blocks, and rather prefer in some cases splitting the code block in two and using normal text between the blocks. IMO that is a bit more readable in the actual html docs (but of course, it's a bit a balance)

**Comment by nickeubank** [comment]
Updated! Will move comments out of ipython more in the future.

**Comment by jorisvandenbossche** [comment]
Thanks a lot!

**Pull request merged**